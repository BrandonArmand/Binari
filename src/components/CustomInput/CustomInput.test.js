import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import CustomInput from "./CustomInput";

describe("CustomInput", () => {
    it("renders snapshot", () => {
        const { container } = render(<CustomInput />);
        expect(container).toMatchSnapshot();
    });
    it("renders CustomInput", () => {
        render(<CustomInput labelText="labelText" id="input_id" />);
        expect(screen.getByRole("textbox")).toBeInTheDocument();
    });
    it("renders CustomInput and click it", async () => {
        const mockClickFn = jest.fn();
        let inputValue = "test";
        const changeEvent = {
            target: { value: "JavaScript" },
        };
        render(
            <CustomInput
                id="input_id"
                labelText="labelText"
                inputProps={{
                    onChange: mockClickFn,
                    value: inputValue,
                }}
            >
                Input
            </CustomInput>
        );
        expect(screen.getByRole("textbox")).toBeInTheDocument();
        const inputElement = screen.getByRole("textbox");
        expect(inputElement).toBeInTheDocument();
        fireEvent.change(inputElement, changeEvent);
        expect(mockClickFn).toHaveBeenCalledTimes(1);
    });
});
