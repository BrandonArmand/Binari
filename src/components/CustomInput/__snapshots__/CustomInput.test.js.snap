// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root makeStyles-formControl-9"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-2 MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input makeStyles-input-10"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
